samcli/local/docker/lambda_debug_settings.py

```diff
@@ -143,3 +149,12 @@ def get_debug_settings(debug_port, debug_args_list, _container_env_vars, runtime
                 LOG.debug("Passing entrypoint as specified in template")
                 return DebugSettings(entry + debug_args_list, _container_env_vars)
             raise DebuggingNotSupported("Debugging is not currently supported for {}".format(runtime)) from ex
+
+    @staticmethod
+    def parse_go_delve_api_version(debug_args_list: List[str]) -> int:
+        parser = ArgumentParser("Parser for delve args")
+        parser.add_argument("-delveApi", type=int, default=1)
+        args, unknown_args = parser.parse_known_args(debug_args_list)
+        if unknown_args:
+            LOG.info("Ignoring unrecognized arguments: %s", unknown_args)
+        return cast("int", vars(args).get("delveApi"))
```

Comment:
More Pythonic:

```python
args.delveApi
```

Reply:
Also I thought `cast` takes a type? i.e. `cast(int, x)`... or the no-op `int(x)`.

Reply:
> args.delveApi

Reply:
> args.delveApi

weird, yesterday I tried that it did not work as if `delveApi` was not an attribute, and now it worked, amazing. Thanks

> Also I thought cast takes a type? i.e. cast(int, x)... or the no-op int(x).

`cast` is a noop (does nothing) while `int()` actually does something. cast() should be faster because it does literally nothing at runtime. Also `'int'` and `int` is effectively the same for mypy (in most cases), changing to `int` to avoid confusion.

> mypy casts are only used as hints for the type checker, and they don’t perform a runtime type check https://mypy.readthedocs.io/en/stable/casts.html

> Return an integer object constructed from a number or string x, or return 0 if no arguments are given. https://docs.python.org/3/library/functions.html#int

Reply:
>I tried that it did not work as if delveApi was not an attribute, and now it worked, amazing

Maybe that was before deconstructing the output? As the parsing function returns a tuple.

LGTM for the `cast`.